receiver/httpreceiver/config.go

```diff
@@ -0,0 +1,68 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package httpreceiver is a default HTTP receiver for log ingestion
+package httpreceiver
+
+import (
+	"errors"
+	"net"
+	"path"
+
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.uber.org/multierr"
+)
+
+// Config defines the configuration for an HTTP receiver
+type Config struct {
+	Endpoint string                      `mapstructure:"endpoint"`
+	Path     string                      `mapstructure:"path"`
+	TLS      *configtls.TLSServerSetting `mapstructure:"tls"`
+}
```

Comment:
Any reason not to embed HTTPServerSettings? 
https://github.com/open-telemetry/opentelemetry-collector/blob/0482753f5203952be14d07be0cd4a45a571d47ec/config/confighttp/confighttp.go#L214

It looks like it has some pretty robust handling of stuff like max body size and auth extensions when using the ToServer function:
https://github.com/open-telemetry/opentelemetry-collector/blob/0482753f5203952be14d07be0cd4a45a571d47ec/config/confighttp/confighttp.go#L289

Reply:
No, not that i'm aware of. I'll change it up to use that.